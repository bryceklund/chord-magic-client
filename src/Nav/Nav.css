h1:hover {
  animation: animation-hero 1s linear infinite;
  background-size: 800% 800%;
  background-image: repeating-linear-gradient(to left, violet, indigo, blue, green, yellow, orange, red);
  color: transparent;
  -webkit-background-clip: text;
  background-clip: text;
}

.logout {
  background: none;
  font-size: 15px;
  padding: 2px;
  border: none;
  color: darkred;
}

.logout:hover {
  color: red;
}

.logout:active {
  color: black;
}

.login_link {
  font-style: none;
}

.login_link:hover {
  color: white;
}

.login_link:active {
  color: gray;
}

.logo {
  width: 40px;
}

.logo_container {
  position: absolute;
}

.logo:hover {
  animation: animation-logo .5s ease-out 1;
  filter: opacity(70%)
}

.logo:active {
  filter: opacity(200%)
}

@keyframes animation-hero {
  0% {
    background-position: right;
  }
  100% {
    background-position: left;
  }
}

@keyframes animation-logo {
  0% {
      transform: rotate(0deg);
  }
  100% {
      transform: rotate(360deg);
  }
}

@media screen and (max-width: 700px) {
  .loggedin_links {
    font-size: 15px;
    display: flex;
    flex-direction: column;
    justify-content: right;
    text-align: right;
  }

  nav {
    padding-left: 5px !important;
    padding-right: 5px !important;
    justify-content: space-between !important;
  }

  .logo_container {
    position: static;
  }
}

@media screen and (min-width: 700px) {
  .login_link {
    font-size: 18px;
  }
}