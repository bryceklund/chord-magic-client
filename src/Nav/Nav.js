import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './Nav.css';


class Nav extends Component {
    render() {
        const signedInFalse = (<div className='login_links'>
                                    <Link to='/login'>Login</Link>&nbsp;
                                    <Link to='/register'>Register</Link>
                               </div>)
        const signedInTrue = (<div className='login_links'>
                                    <Link to='/saved'>My Progressions</Link>&nbsp;
                                    <Link to='/logout'>Logout</Link>
                              </div>)
        return (
            <nav>   
                <h1><Link to='/editor'>Chord Magic</Link></h1>
                <Link to='/editor'><img src='' alt='chord magic logo' /></Link>
                {!this.props.signedIn ? signedInFalse : signedInTrue}
            </nav>
        );
    }
}

export default Nav;