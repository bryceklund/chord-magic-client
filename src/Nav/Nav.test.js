import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import Nav from './Nav'

describe('Nav', () => {
    it(`renders logged in without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter>
                            <Nav signedIn={true} />
                        </BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })
    it(`renders logged out without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter>
                            <Nav signedIn={false} />
                        </BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })
    it(`matches its snapshot`, () => {
        const tree = renderer.create(<BrowserRouter>
                                        <Nav />
                                    </BrowserRouter>)
                                .toJSON()
        expect(tree).toMatchSnapshot()
    })
})