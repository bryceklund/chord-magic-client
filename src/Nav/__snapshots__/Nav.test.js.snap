// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav matches its snapshot 1`] = `
<nav>
  <h1>
    <a
      href="/editor"
      onClick={[Function]}
    >
      Chord Magic
    </a>
  </h1>
  <a
    href="/editor"
    onClick={[Function]}
  >
    <img
      alt="chord magic logo"
      src=""
    />
  </a>
  <div
    className="login_links"
  >
    <a
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
    Â 
    <a
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
  </div>
</nav>
`;
